modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/JobDispatcher.java

```diff
@@ -0,0 +1,646 @@
+          for (JpaJob job : dispatchableJobs) {
+            if (ServiceRegistryJpaImpl.TYPE_WORKFLOW.equals(job.getJobType())) {
+              workflowJobs.add(job);
+            }
+          }
+          if (dispatchableJobs.removeAll(workflowJobs) && dispatchableJobs.isEmpty()) {
+            continue;
+          }
+
+          dispatchDispatchableJobs(em, dispatchableJobs);
+        } while (jobsFound);
+
+        if (!workflowJobs.isEmpty()) {
+          dispatchDispatchableJobs(em, workflowJobs);
+        }
+
+      } catch (Throwable t) {
+        logger.warn("Error dispatching jobs", t);
+      } finally {
+        undispatchableJobTypes = null;
+        if (em != null) {
+          em.close();
+        }
+      }
+
+      logger.debug("Finished job dispatch");
+    }
+
+    /**
+     * Dispatch the given jobs.
+     *
+     * @param em             the entity manager
+     * @param jobsToDispatch list with dispatchable jobs to dispatch
+     */
+    private void dispatchDispatchableJobs(EntityManager em, List<JpaJob> jobsToDispatch) {
+      //Get the current system load
+      SystemLoad systemLoad = serviceRegistry.getHostLoads(em);
+
+      for (JpaJob job : jobsToDispatch) {
+
+        // Remember the job type
+        String jobType = job.getJobType();
+
+        // Skip jobs that we already know can't be dispatched except of jobs in the priority list
+        String jobSignature = new StringBuilder(jobType).append('@').append(job.getOperation()).toString();
+        if (undispatchableJobTypes.contains(jobSignature) && !dispatchPriorityList.keySet().contains(job.getId())) {
+          logger.trace("Skipping dispatching of {} with type '{}' for this round of dispatching", job, jobType);
+          continue;
+        }
+
+        // Set the job's user and organization prior to dispatching
+        String creator = job.getCreator();
+        String creatorOrganization = job.getOrganization();
+
+        // Try to load the organization.
+        Organization organization = null;
+        try {
+          organization = organizationDirectoryService.getOrganization(creatorOrganization);
+          securityService.setOrganization(organization);
+        } catch (NotFoundException e) {
+          logger.debug("Skipping dispatching of job for non-existing organization '{}'", creatorOrganization);
+          continue;
+        }
+
+        // Try to load the user
+        User user = userDirectoryService.loadUser(creator);
+        if (user == null) {
+          logger.warn("Unable to dispatch {}: creator '{}' is not available", job, creator);
+          continue;
+        }
+        securityService.setUser(user);
+
+        // Start dispatching
+        try {
+          List<ServiceRegistration> services = serviceRegistry.getServiceRegistrations(em);
+          List<HostRegistration> hosts = Stream.$(serviceRegistry.getHostRegistrations(em)).filter(filterOutPriorityHosts._2(job.getId())).toList();
```

Comment:
Non-Eentwine streams?

Reply:
Copy and paste from the old codebase for the win.  Will fix.